readme.md

```diff
@@ -45,6 +45,21 @@ Because this proposal allows existing contracts to remove higher-precision math
 ## Technical Specification
 
 The limit on the maximum length of Bitcoin Cash VM numbers (A.K.A. `nMaxNumSize`) is removed.
+The numbers will still be limited, but by the maximum stack item size (A.K.A. `MAX_SCRIPT_ELEMENT_SIZE`).
+The effective limits will then be a calculated value from `MAX_SCRIPT_ELEMENT_SIZE`:
+
+- `MAX_SCRIPTNUM = (MAX_SCRIPT_ELEMENT_SIZE * 8 - 1) - 1`, and
+- `MIN_SCRIPTNUM = -MAX_SCRIPTNUM`,
+
+rather than independently set limits (as they were before this upgrade).
```

Comment:
This is an implementation detail, not part of the spec. It's ok to explain it as an implementation detail, but it's just an example of how one might use it.

Also if you are going to establish it as an implementation detail, you need to show that it is not possible to construct a number `x` such that x is outside the range `[MAX_SCRIPTNUM, MIN_SCRIPTNUM]` while still fitting in the byte size of `MAX_SCRIPT_ELEMENT_SIZE`.

Reply:
i.e. it's not enough to say that the range is `[-2^79999 + 1, 2^79999 - 1]`. That needs to be shown in binary format of VM numbers that it's not possible to construct something that breaks this linkage.

Reply:
Which adds more complexity and why I like the elegance of just leaving it to the byte size. One metric. No room for error.

Reply:
how's this: https://github.com/bitjson/bch-bigint/pull/8/commits/051951a508b3736fa51a274b6430386172d2bc57

Reply:
after the calculation above gets fixed, yeah it all fits together pretty well now :+1: